lexicons/com/atproto/repo/getRecord.json

```diff
@@ -1,27 +1,27 @@
 {
   "lexicon": 1,
   "id": "com.atproto.repo.getRecord",
-  "type": "query",
-  "description": "Fetch a record.",
-  "parameters": {
-    "type": "object",
-    "required": ["user", "collection", "rkey"],
```

Comment:
Current feeling is that I think I'll miss "required" for params.  I don't recall what the considerations there were, though.

Reply:
Bleh. (Not you, the situation.) Parameters are awkward because they're almost exactly like objects but they're not, so we can't reuse object definitions but it's hard to deviate from them too. The loss of "required" was me making the least awkward choice for the schema language.

I'm open to ideas for re-introducing that information.

Reply:
I don't think object is so far off, especially now that you don't allow sub-objects in `object` unless they are behind a reference.  We sort of have the same issue with json, since it could contain a ref to something it can't support, such as a `video`.  My gut would be to treat them as objects.

Side note—the fact that arrays are allowed inside object also seems alright to me.  While params don't support sub-objects in any standard way, they do support arrays: `x=one&x=two` is `x: ['one', 'two']` (can see this reflected e.g. in [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams)).

Reply:
99% you know this, but just to clarify a bit --

"Lexicon refs" and "data links" are two different things. Lexrefs are essentially links in lexicon definitions, while data links are links in values. Objects _do_ allow sub-objects, but lexicon object definitions _dont_ allow inline object definitions -- you have to use a lexref. So: more accurately, LexParams are LexObject subsets that don't allow sub-LexObjects.

I suppose a simple solution would be to mirror the object-definition structure accordingly:

```js
"parameters": {
  "type": "params",
  "required": ["foo"],
  "params": { // or "properties"
    "foo": {"type": "string"},
    "bar": {"type": "string"},
  }
}
```

It's more verbose, but it's also more consistent and it gives us a place to put the required-param info.

Reply:
Ah yeah sorry, I did get a little turned around there in a couple ways.  Ignore the `video` example!  I was also looking at the spec which shows this for `object`:
```js
properties: Record<string, LexRef | LexArray | LexPrimitive | LexRef[]>
```

I see now that the blob variants are also allowed there, though ✨ 

The main thought I was having is that objects can already contain refs to things that they can't represent, and that is the same issue we have with params.  I am realizing now that you've spec'ed that issue away for objects, though.  So the issue that remains is just that parameters could contain a ref to an object, and that's not allowed.  I think the proposal you tossed out seems very reasonable 👍 👍